sdks/java/harness/src/main/java/org/apache/beam/fn/harness/MapFnRunner.java

```diff
@@ -45,31 +45,45 @@
  * <p>Simple map functions are used in a large number of transforms, especially runner-managed
  * transforms, such as map_windows.
  *
- * <p>TODO: Add support for DoFns which are actually user supplied map/lambda functions instead
- * of using the {@link FnApiDoFnRunner} instance.
+ * <p>TODO: Add support for DoFns which are actually user supplied map/lambda functions instead of
+ * using the {@link FnApiDoFnRunner} instance.
  */
 public class MapFnRunner<InputT, OutputT> {
 
-  public static <InputT, OutputT> PTransformRunnerFactory<?>
-      createMapFnRunnerFactoryWith(
-          CreateMapFunctionForPTransform<InputT, OutputT> fnFactory) {
+  public static <InputT, OutputT> PTransformRunnerFactory<?> forValueMapFnFactory(
+      ValueMapFnFactory<InputT, OutputT> fnFactory) {
+    return forMapFnFactory(new ValueOnlyWindowedValueMapFnFactory<>(fnFactory));
+  }
+
+  public static <InputT, OutputT> PTransformRunnerFactory<?> forMapFnFactory(
```

Comment:
Add a comment stating that the windowed value will be exploded into each window separately.